From ec59128bb0a2fee53092e1c8bd9934cc63ab88f2 Mon Sep 17 00:00:00 2001
From: Reto Schneider <reto.schneider@husqvarnagroup.com>
Date: Tue, 27 Apr 2021 13:22:38 +0200
Subject: [PATCH] [sg fromtree] shared library: Link against zlib

Library is using crc32(), therefore needs to link against zlib.

Signed-off-by: Reto Schneider <reto.schneider@husqvarnagroup.com>
Reviewed-by: Stefano Babic <sbabic@denx.de>
(cherry picked from commit f4ab25ded1b92b9a260253ad3a9e8722b4f23797)
---
 src/CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a04dd11..4b71bc5 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -19,8 +19,9 @@ ADD_LIBRARY(ubootenv_static STATIC ${libubootenv_SOURCES} ${include_HEADERS})
 SET_TARGET_PROPERTIES(ubootenv_static PROPERTIES OUTPUT_NAME ubootenv)
 add_executable(fw_printenv fw_printenv.c)
 add_executable(fw_setenv fw_setenv.c)
-target_link_libraries(fw_printenv ubootenv z)
-target_link_libraries(fw_setenv ubootenv z)
+target_link_libraries(ubootenv z)
+target_link_libraries(fw_printenv ubootenv)
+target_link_libraries(fw_setenv ubootenv)
 
 install (TARGETS ubootenv ubootenv_static DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install (FILES libuboot.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-- 
2.30.2

